src/studio/src/designer/backend.Tests/_TestData/Model/Metadata/HvemErHvem.metadata.json

```diff
@@ -0,0 +1 @@
","parentElement":"melding","typeName":"Innrapportoer","name":"Innrapportoer","dataBindingName":null,"xPath":"/melding/Innrapportoer","restrictions":{},"choices":null,"type":"Group","xsdValueType":null,"texts":{},"customProperties":{},"maxOccurs":1,"minOccurs":1,"xName":"Innrapportoer","isTagContent":false,"fixedValue":null,"isReadOnly":false,"xmlSchemaXPath":null,"jsonSchemaPointer":"#/definitions/HvemErHvem_M/properties/Innrapportoer","displayString":"melding.Innrapportoer : [1..1] Innrapportoer"},"melding.InnrapporterteData.geekType":{"id":"melding.InnrapporterteData.geekType","parentElement":"melding.InnrapporterteData","typeName":"GeekType","name":"geekType","dataBindingName":"InnrapporterteData.geekType","xPath":"/melding/InnrapporterteData/geekType","restrictions":{"enumeration":{"value":"frontend;backend;other","errortText":null}},"choices":null,"type":"Field","xsdValueType":"String","texts":{},"customProperties":{},"maxOccurs":1,"minOccurs":1,"xName":"geekType","isTagContent":false,"fixedValue":null,"isReadOnly":false,"xmlSchemaXPath":null,"jsonSchemaPointer":"#/definitions/InnrapporterteData/properties/geekType","displayString":"melding.InnrapporterteData.geekType : [1..1] GeekType"},"melding.InnrapporterteData.altinnErfaringAAr":{"id":"melding.InnrapporterteData.altinnErfaringAAr","parentElement":"melding.InnrapporterteData","typeName":"AltinnErfaringAAr","name":"altinnErfaringAAr","dataBindingName":"InnrapporterteData.altinnErfaringAAr","xPath":"/melding/InnrapporterteData/altinnErfaringAAr","restrictions":{},"choices":null,"type":"Field","xsdValueType":"Integer","texts":{},"customProperties":{},"maxOccurs":1,"minOccurs":0,"xName":"altinnErfaringAAr","isTagContent":false,"fixedValue":null,"isReadOnly":false,"xmlSchemaXPath":null,"jsonSchemaPointer":"#/definitions/InnrapporterteData/properties/altinnErfaringAAr","displayString":"melding.InnrapporterteData.altinnErfaringAAr : [0..1] AltinnErfaringAAr"},"melding.InnrapporterteData":{"id":"melding.InnrapporterteData","parentElement":"melding","typeName":"InnrapporterteData","name":"InnrapporterteData","dataBindingName":null,"xPath":"/melding/InnrapporterteData","restrictions":{},"choices":null,"type":"Group","xsdValueType":null,"texts":{},"customProperties":{},"maxOccurs":1,"minOccurs":1,"xName":"InnrapporterteData","isTagContent":false,"fixedValue":null,"isReadOnly":false,"xmlSchemaXPath":null,"jsonSchemaPointer":"#/definitions/HvemErHvem_M/properties/InnrapporterteData","displayString":"melding.InnrapporterteData : [1..1] InnrapporterteData"},"melding":{"id":"melding","parentElement":null,"typeName":"HvemErHvem_M","name":"melding","dataBindingName":null,"xPath":"/melding","restrictions":{},"choices":null,"type":"Group","xsdValueType":null,"texts":{},"customProperties":{},"maxOccurs":1,"minOccurs":1,"xName":"melding","isTagContent":false,"fixedValue":null,"isReadOnly":false,"xmlSchemaXPath":null,"jsonSchemaPointer":"#/properties/melding","displayString":"melding : [0..1] HvemErHvem_M"}}}
```

Comment:
Not sure how to set the typeName value when the type is undefined. I'm thinking of the cases like for EPost. We reference EPost as typeName, but EPost isn't defined. I see you also comment in the MetadataAssertion logic that typeName might be wrong in expected.

Reply:
Yes, that puzzled me as well when I looked at how this is currently implemented. I would have expected the type name to be string in this case, but then we're not generating the same data as we used to.